src/main/scala/com/kinja/soy/SoyMacroImpl.scala

```diff
@@ -173,4 +173,174 @@ object SoyMacroImpl {
     }
   }
 
+  def typedWritesImpl[A](c: Context)(typeValue: c.Expr[String])(implicit atag: c.WeakTypeTag[A]): c.Expr[SoyMapWrites[A]] = {
```

Comment:
90% of this implementation is the same as `writesImpl` and aside from the extra field they should always produce identical output. Can you move the implementation into a shared function that takes a `typed : Boolean` to determine whether to add the extra field?